### ISSUE TITLE
Propagate the return value from a script as the process exit code

### ISSUE THREAD

**Issue by adamralph** [filipw/dotnet-script#158]
With csi.exe, the `int` returned from a script (it is only allowed to be an `int`) is propagated as the process exit code. E.g. if I return `0` from my script, csi.exe exits with `0` which indicates no error. If I return `1` from my script, `csi.exe` exits with `1`, which indicates an error.

`dotnet-script` should match this behaviour from `csi.exe`. Specifically, we need the runner to behave in this manner to support this proposal for simple-targets-csx: https://github.com/adamralph/simple-targets-csx/issues/49

**Comment by seesharper** [comment]
Makes sense to me :+1 Is there a specific code for compilation errors that we should make sure to match?

**Comment by adamralph** [comment]
This isn't for compilation errors. This is for when the script compiles, and runs to completion with no exception being thrown. At the end of the script an integer is returned, and that integer is propagated to the OS as the exit code of dotnet-script.

**Comment by adamralph** [comment]
E.g.

```C#
// foo.csx

// do some stuff

if (everythingWorkedOK)
{
    return 0;
}
else
{
    return 1;
}
```

**Comment by filipw** [comment]
For compilation errors CSI returns 1.

**Comment by seesharper** [comment]
> This isn't for compilation errors.

I got that :) Was just making sure that we align the exit code for compilation errors as well.

**Comment by adamralph** [comment]
OK 😉 

Bear in mind that the 1 returned by csi.exe could be a compilation error, or it could be a 1 returned from a script, so it doesn't really give you any information. Sure, you could attempt to parse stderr and sniff out the compilation error from there, but then perhaps the script itself is using the roslyn libs. Very difficult to be deterministic there.

**Comment by filipw** [comment]
Closed by #160
